src/Sulu/Component/Rest/ListBuilder/Metadata/ProviderInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Metadata;
+
+use Metadata\ClassMetadata;
+
+/**
+ * Interface for metadata-providers.
+ */
+interface ProviderInterface
```

Comment:
each provider is able to extend the metadata with its own information the metadata-factory is used by the provider to read the xml file.